src/qibo/quantum_info/metrics.py

```diff
@@ -385,3 +385,92 @@ def gate_error(channel, target=None, backend=None):
     error = 1 - average_gate_fidelity(channel, target, backend=backend)
 
     return error
+
+
+def meyer_wallach(circuit, backend=None):
+    """Computes the Meyer-Wallach entanglement Q of the `circuit`,
+
+    .. math::
+        Q = 1-\\frac{1}{N}\\sum_{k}\\text{Tr}\\left(\\rho_k^2(\\theta_i)\\right) \\,
+
+    Args:
+        circuit (qibo.models.Circuit): Parametrized circuit.
+        backend (qibo.backends.abstract.Backend): Calculation engine.
+
+    Returns:
+        (int) : Meyer-Wallach entanglement.
+    """
+
+    if backend == None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    circuit.density_matrix = True
+    nqubits = circuit.nqubits
+    rho = backend.execute_circuit(circuit).state()
+    entropy = 0
+    for j in range(nqubits):
+        trace_q = list(range(nqubits))
+        trace_q.pop(j)
+        rho_r = backend.partial_trace_density_matrix(rho, trace_q, nqubits)
+        trace = np.trace(rho_r @ rho_r)
+        entropy += trace
+    return 1 - entropy / nqubits
+
+
+def entangling_capability(circuit, samples, backend=None):
```

Comment:
Also following the rest of the module, preferably use type hints when types are basic types (e.g. int, float, str, bool, etc)